src/models/transfers.js

```diff
@@ -256,17 +257,26 @@ function isAuthorized (transfer) {
   return areDebitsAuthorized
 }
 
-function * processTransitionToPreparedState (transfer, accountBalances) {
+function * processTransitionToPreparedState (transfer, transaction) {
   if (transfer.state === transferStates.TRANSFER_STATE_PROPOSED && isAuthorized(transfer)) {
-    yield accountBalances.applyDebits()  // hold sender funds
+    try {
+      yield holds.holdFunds(transfer, transaction)  // hold sender funds
+    } catch (e) {
+      if (e.code === 'SQLITE_CONSTRAINT') {
```

Comment:
Is there any way we can get a more specific error? Presumably this won't work for oracle or sql server.